generators/handlebars/index.js

```diff
@@ -89,94 +89,11 @@ module.exports = class extends Generator {
     }
 
     _addPackageDependencies() {
```

Comment:
OK.

Reply:
The types, that you're talking about are types definitions for Vue itself.
But typescript still doesn't know that *.vue files are modules.
You need either add a definition file for *.vue files or change ts loader settings in webpack config to add .ts extension to .vue files. 
I have selected the first one.
If you delete this file you'll get tsc error:
<img width="1322" alt="screen shot 2018-09-20 at 10 06 25 am" src="https://user-images.githubusercontent.com/17036219/45834812-fda32080-bcbc-11e8-8f03-6c45dd69a58b.png">

Reply:
Your problem now is completely unrelated to the types. You cannot load a `.vue` file in your a Typescript file because it is not a valid JavaScript or TypeScript file. I know there are some tutorials on the web that really import Vue but those will get compiled through BabelJS or TypeScript to file names like
`mycomponent.vue.js` in case of Babel or `mycomponent.vue.ts` in case of TypeScript.
In your case it is not working because there is no compiler in place. Before you imported the `.ts` file, which obviously worked but it also means that all VueJS component needed to be written in TypeScript.
The Vue loader on the other hand is smart enough to know how to deal with your Vue file but only because of your TypeScript file. If a type definition would have been missing you would get a complete different error.
I my case there are no official types available for Handlebars. I need to install them separately through `npm install @types/handlebars`.
You could do the same through `@types/vue` rather than place this file in your project but there is another problem. This type definition is deprecated https://www.npmjs.com/package/@types/vue because it is already included.
Can you please do me a favor and please check what needs to be configured in Webpack and with VueJS to compile the templates to a [Typescript](https://vuejs.org/v2/guide/typescript.html) file properly aka [pre-compiled templates](https://vuejs.org/v2/guide/deployment.html#Pre-Compiling-Templates)
In addition it should compile the CSS through SASS and should support [CSS Modules](https://vue-loader.vuejs.org/guide/css-modules.html).
Right now the VueJS configuration is incomplete and work not optimal because people have certain exceptions when we announce the VuejS is now part of this project. Thank you very much for your understanding.

Reply:
I know that it’s unrelated to the types. And I was telling it from the beginning, that shims file is needed to tell TypeScript that .vue file is a module.

Reply:
Not in this case and not when you do not compile the Typescript out of the SFC.

Reply:
shims is needed for compiler to "know" what is exported from .vue file.